File: Projects/TransportathonHackathon.Clients/angular-ui/src/app/models/domain-models/message.ts
import { Entity } from 'src/app/core/models/entity';
import { AppUser } from './appUser';

export interface Message extends Entity<string> {
  senderId: string;
  receiverId: string;
  messageText: string;
  sendDate: Date;
  isRead: boolean;
  sender: AppUser;
  receiver: AppUser;
}

File: Projects/TransportathonHackathon.API/TransportathonHackathon.Domain/Entities/Message.cs
using Core.Persistence.Entities;
using TransportathonHackathon.Domain.Entities.Identity;

namespace TransportathonHackathon.Domain.Entities
{
    public class Message : Entity<Guid>
    {
        public Guid SenderId { get; set; }
        public Guid ReceiverId { get; set; }
        public string MessageText { get; set; }
        public DateTime SendDate { get; set; }
        public bool IsRead { get; set; }
        public int Queue {  get; set; }

        public virtual AppUser Sender { get; set; }
        public virtual AppUser Receiver { get; set; }
    }
}
